src/App.jsx

```diff
@@ -2,13 +2,44 @@ import "../src/assets/css/common.css";
 import "../src/assets/css/details.css";
 import "../src/assets/css/main.css";
 import "../src/assets/scss/common.scss";
-import Home from "./pages/Home";
 
+import Home from "./Home.jsx";
+import {useState} from "react";
+
+/**
+ * App Component
+ * Manages the application, including theme state and persistence.
+ *
+ * State:
+ * - theme: Tracks the current theme ("light" or "dark"), initialized from localStorage.
+ *
+ */
 function App() {
+    const [theme, setTheme] = useState(() => {
+        const savedTheme = localStorage.getItem("theme");
+
+        if (savedTheme) {
+            document.body.classList.toggle("dark-theme", savedTheme === "dark");
+            return savedTheme;
```

Comment:
If the user does not have any theme why not give default theme? 

This way you will not have to check the value of saved theme